crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__statement__try_py.snap

```diff
@@ -0,0 +1,110 @@
+---
+source: crates/ruff_python_formatter/src/lib.rs
+expression: snapshot
+---
+## Input
+```py
+try:
+    ...
+except:
+    ...
+
+
+try: # try
+    ...
+    # end of body
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+# before except 2
+except (KeyError) as key:  # except line 2
+    ...
+    # in body 2
+# before else
+else:
+    ...
+# before finally
+finally:
+    ...
+
+
+
+# with line breaks
+try: # try
+    ...
+    # end of body
+
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+
+# before except 2
+except (KeyError) as key:  # except line 2
+    ...
+    # in body 2
+
+# before else
+else:
+    ...
+
+# before finally
+finally:
+    ...
+```
+
+
+
+## Output
+```py
+try:
+    ...
+except:
+    ...
+
+
+try:
+    # try
+    ...
+    # end of body
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+# before except 2
+except (
+    KeyError  # except line 2
+) as key:
+    ...
+    # in body 2
+# before else
+else:
+    ...
+# before finally
+finally:
+    ...
+
+
+# with line breaks
+try:
+    # try
+    ...
+    # end of body
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+# before except 2
+except (
+    KeyError  # except line 2
```

Comment:
as far as i can see this is currently the only misplaced comment, right?

black also seems to remove the parentheses here even though it otherwise doesn't

Reply:
comment placement looks ok to me here (`except line 2`)

but i don't understand what causes us to break the `(KeyError)` over multiple lines

Reply:
note that this is just a call to `.format()`, on a tuple of types, nothing manual here 
https://github.com/astral-sh/ruff/pull/5222/files/ccd96e6ae3d7e38b833f530d99f5d0b9d46b35ef#diff-30bdf42955f6a58b1f41492d192056475d73d2d27906cb200ba4beba5d010d75R39

Reply:
black format
```python
try:
    ...
except (KeyError) as key:  # except line 2
    ...
```
as
```
try:
    ...
except KeyError as key:  # except line 2
    ...
```
which looks better to me. This is another one of the rare cases where black removes parentheses

Reply:
i agree that it's better to remove them. but i'm not sure what is preserving them (i was wrong above, here it isn't a tuple but just an `ExprName`)

Reply:
You might need to change the attachment of `# except line 2`  in placement.rs, either attach it trailing to a node that doesn't break the line if it doesn't need to or mark it as dangling and manually format after the colon. we're doing the latter (formatting after `:` manually) in some other nodes

Reply:
Yeah... The fact that there's no explicit node for bodies really makes things annoying. The issue here is that the `except line 2` comment falls into the middle of the `ExceptHandlerExceptHandler` node (it is inside of that node's range). The last preceding node is the `KeyError` because `key` (`Identifier`) is not an AST node (it probably should, but it is not :cry:). 

We probably need something similar to (or extend) https://github.com/astral-sh/ruff/blob/8a73a7b481b11d161a7ba2f1b56c59ca977f0848/crates/ruff_python_formatter/src/comments/placement.rs#L551-L554